backend/api/routes/files.py

```diff
@@ -52,74 +61,145 @@ def validated_file(
+    """Validates total size of uploaded files to ensure it meets the requirements."""
+    total_size = sum([file.filesize for file in project.files]) + file_size
+    if total_size > BackendConf.maximum_project_quota:
+        raise HTTPException(
+            status_code=codes.REQUEST_ENTITY_TOO_LARGE,
+            detail="Uploaded files exceeded quota",
+        )
+
+
+@router.post("/{project_id}/files", status_code=codes.CREATED)
+async def create_file(
+    uploaded_file: UploadFile,
     project: Project = Depends(validated_project),
     session: Session = Depends(gen_session),
-) -> list[FileModel]:
-    """Upload new files"""
-    added_files = []
-    temp_file_location = BackendConf.cache_path.joinpath("files")
-    for file in uploaded_files:
-        binary_file = await file.read()
-
-        location = upload_file(temp_file_location, binary_file)
-
-        now = datetime.datetime.now(tz=datetime.UTC)
-        filename = file.filename if file.filename else ""
-        size = file.size if file.size else 0
-        content_type = file.content_type if file.content_type else "text/plain"
-        new_file = File(
-            filename=filename,
-            filesize=size,
-            title=filename,
-            authors=None,
-            description=None,
-            uploaded_on=now,
-            hash=hashlib.sha256(binary_file).hexdigest(),
-            path=str(location.resolve()),
-            type=content_type,
-            status="LOCAL",
-        )
-        project.files.append(new_file)
-        session.add(new_file)
-        session.flush()
-        session.refresh(new_file)
-        added_files.append(new_file)
-    return TypeAdapter(list[FileModel]).validate_python(added_files)
+) -> FileModel:
+    """
+    Uploads a new file and creates a corresponding FileModel.
+
+    Returns:
+
+        FileModel: The created FileModel.
+
+    Note:
+
+        HTTPException(406, "Filename is invalid"): If the filename is invalid.
+
+        HTTPException(406, "Emtpy File"): the file is empty.
+
+        HTTPException(416, "Uploaded files exceeded quota"):
+            the file size exceeds the maximum allowed size.
+    """
+    now = datetime.datetime.now(tz=datetime.UTC)
+
+    filename, size, mimetype = validate_uploaded_file(uploaded_file)
+    validate_project_quota(size, project)
+
+    location = upload_file(BackendConf.temp_files_location, uploaded_file.file)
+
+    new_file = File(
+        filename=filename,
+        filesize=size,
+        title=filename,
+        authors=None,
+        description=None,
+        uploaded_on=now,
+        hash=generate_file_hash(uploaded_file.file),
+        # TODO: Using S3 to save file.
+        path=str(location.resolve()),
+        type=mimetype,
+        status=FileSaveLocation.LOCAL.value,
+    )
+    project.files.append(new_file)
+    session.add(new_file)
+    session.flush()
+    session.refresh(new_file)
+    return FileModel.model_validate(new_file)
```

Comment:
Code should be `201`

Reply:
`status_code` is `codes.CREATED(201)` now. I will change it to `http` lib.